graphrag/config/load_config.py

```diff
@@ -49,17 +50,20 @@ def load_config(
 
     if run_id:
         config.storage.base_dir = str(
-            resolve_timestamp_path((root / config.storage.base_dir).resolve(), run_id)
+            resolve_timestamp_path(config.storage.base_dir, run_id)
         )
         config.reporting.base_dir = str(
-            resolve_timestamp_path((root / config.reporting.base_dir).resolve(), run_id)
+            resolve_timestamp_path(config.reporting.base_dir, run_id)
         )
     else:
-        config.storage.base_dir = str(
-            resolve_timestamp_path((root / config.storage.base_dir).resolve())
-        )
+        config.storage.base_dir = str(resolve_timestamp_path(config.storage.base_dir))
         config.reporting.base_dir = str(
-            resolve_timestamp_path((root / config.reporting.base_dir).resolve())
+            resolve_timestamp_path(config.reporting.base_dir)
```

Comment:
Wonder if we can simplify the login in the load_config method by translating some of the responsability to resolve_timestamp_path.
Make it more generic, something like resolve_path that receives the path, the root, the type, the id, etc. And allows for nulls and such.
Also, we can use switch case when possible, it resolves faster than if else.